### PR TITLE
chore: Add deprecation notes to customStateCheck

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Add deprecation notes to customStateCheck.

Changes proposed in this pull request:

- Add
- deprecation note to customStateCheck in the api
- deprecation note to customStateCheck in the docs

**Related issue(s)**
fixes: #1616

### CODE REVIEW

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 1]
```diff
@@ -49,6 +49,8 @@ If not specified, the **namespace** of the resource mentioned in **.spec.data**
 
 ### **.spec.customStateCheck**
 
+**Warning** This field is deprecated and will be removed in the future.
```
Comment (mmitoraj) [inline docs/technical-reference/api/moduleTemplate-cr.md:52]:
```suggestion
> **CAUTION:** This field is deprecated and will be removed in the future. 
```

**Review by mmitoraj — COMMENTED** [review]
I also found mentions of **customStateCheck** in the following files in other repos:
- https://github.com/kyma-project/template-operator?tab=readme-ov-file#build-and-push-your-module-to-the-registry
- https://github.tools.sap/kyma/module-manifests#ii-build-a-kubernetes-operator

**Comment by Tomasz-Smelcerz-SAP** [comment]
@mmitoraj I have created two PRs to remove mentions you found:
- https://github.com/kyma-project/template-operator/pull/227
- https://github.tools.sap/kyma/module-manifests/pull/552

**Review by lindnerby — DISMISSED** [review]

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 2]
```diff
@@ -49,6 +49,8 @@ If not specified, the **namespace** of the resource mentioned in **.spec.data**
 
 ### **.spec.customStateCheck**
 
+> **CAUTION:** This field is deprecated and will be removed in the future.
```
Comment (mmitoraj) [inline docs/technical-reference/api/moduleTemplate-cr.md:52]: Let's provide the deprecation date also here.

Comment (mmitoraj) [inline docs/technical-reference/api/moduleTemplate-cr.md:52]:
```suggestion
> **CAUTION:** This field is deprecated and will be removed on {DATE}. 
```

Comment (mmitoraj) [inline docs/technical-reference/api/moduleTemplate-cr.md:52]:
```suggestion
> **CAUTION:** This field was deprecated at the end of July 2024 and will be removed on {DATE}. 
```
We could also provide the deletion date. Is there any action required from the users?

**Review by mmitoraj — APPROVED** [review]

**Review by lindnerby — APPROVED** [review]

**Pull request merged**